tasks/update_go.py

```diff
@@ -0,0 +1,179 @@
+"""
+    # hiding stderr too because `reno` displays some warnings about the config
+    res = ctx.run(f'reno new "bump go to {version}"', hide='both')
+    match = re.match("^Created new notes file in (.*)$", res.stdout, flags=re.MULTILINE)
+    if not match:
+        raise exceptions.Exit("Could not get created release note path")
+
+    path = match.group(1)
+    with open(path, "w") as writer:
+        writer.write(RELEASENOTE_TEMPLATE.format(version))
+    return path
+
+
+@task(
+    help={
+        "version": "The version of Go to use",
+        "image_tag": "Tag from buildimages with format v<build_id>_<commit_id>",
+        "test_version": "Whether the image is a test image or not",
+        "force_major": "Whether to apply changes like if it was a major version update",
+    }
+)
+def update_go(
+    ctx: Context,
+    version: str,
+    image_tag: str,
+    test_version: Optional[bool] = False,
+    force_major: Optional[bool] = False,
+):
+    """
+    Updates the version of Go and build images.
+    """
+    if not re.match("[0-9]+.[0-9]+.[0-9]+", version):
+        raise exceptions.Exit(
+            "The version doesn't have an expected format, it should be 3 numbers separated with a dot."
+        )
+
+    # check the installed go version before running `tidy_all`
+    res = ctx.run("go version")
+    if not res.stdout.startswith(f"go version go{version} "):
+        raise exceptions.Exit("The version of your `go` binary doesn't match the requested version")
+
+    current_version = _get_repo_go_version()
+    current_major = _get_major_version(current_version)
+    new_major = _get_major_version(version)
+
+    major_update = current_major != new_major or force_major
+    if major_update:
+        print(color_message("WARNING: this is a change of major version\n", "orange"))
+        _update_readme(new_major)
+        _update_go_mods(new_major)
+
+    update_gitlab_config(".gitlab-ci.yml", image_tag, test_version=test_version)
+    update_circleci_config(".circleci/config.yml", image_tag, test_version=test_version)
+    _update_go_version_file(version)
+    _update_gdb_dockerfile(version)
+    _update_install_devenv(version)
+    _update_agent_devenv(version)
+    _update_task_go(version)
+
+    tidy_all(ctx)
+
+    releasenote_path = _create_releasenote(ctx, version)
+    print(
+        f"A default release note was created at {releasenote_path}, edit it if necessary, for example to list CVEs it fixes."
+    )
+    if major_update:
+        # Examples of major updates with long descriptions:
+        # releasenotes/notes/go1.16.7-4ec8477608022a26.yaml
+        # releasenotes/notes/go1185-fd9d8b88c7c7a12e.yaml
+        print("In particular as this is a major update, the release note should describe user-facing changes.")
+    print()
+
+    print(
+        color_message(
+            "Remember to look for reference to the former version by yourself too, and update this task if you find some.",
```

Comment:
Does this sentence miss a word after `some`?

Reply:
I guess `any` would have been better but no, it's implicitly "if you find any reference to the former version"
If you have a better sentence I can update it